core/kernel/tee_ta_manager.c

```diff
@@ -101,6 +101,15 @@ static bool tee_ta_try_set_busy(struct tee_ta_ctx *ctx)
 
 	mutex_lock(&tee_ta_mutex);
 
+	if (ctx->initializing) {
+		/*
+		 * Context is still initializing and flags cannot be relied
+		 * on for user TAs. Wait here until it's initialized.
+		 */
+		while (ctx->busy)
+			condvar_wait(&ctx->busy_cv, &tee_ta_mutex);
```

Comment:
Why are we guaranteed to have `ctx->initializing == false` after the wait loop on `ctx->busy`?

Reply:
Because if `ctx->busy` then `ctx->initializing` will be cleared at the same time as `ctx->busy` is cleared. On the other hand if not `ctx->busy` then we're supposed to continue but with some special treatment in `tee_ta_clear_busy()` since `ctx->flags` has been updated.